tests/test_username_api.py

```diff
@@ -351,94 +108,17 @@ def test_available_deviantart_username(self, user):
-    } == json_resp
-
-  def test_gitlab_format_checking(self):
-    resp = self.app.get('/check/gitlab/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://gitlab.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_tumblr_format_checking(self):
-    resp = self.app.get('/check/tumblr/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://{}.tumblr.com'.format(invalid_username)
-    } == json_resp
-
-  def test_behance_format_checking(self):
-    resp = self.app.get('/check/behance/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://behance.net/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_pinterest_format_checking(self):
-    resp = self.app.get('/check/pinterest/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://in.pinterest.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_instagram_format_checking(self):
-    resp = self.app.get('/check/instagram/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://instagram.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_twitter_format_checking(self):
-    resp = self.app.get('/check/twitter/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://twitter.com/{}'.format(invalid_username)
-    } == json_resp
-
-  def test_facebook_format_checking(self):
-    invalid_username_facebook = ['hello..world', 'hell...', 'manvendra.com']
-
-    for invalid_username in invalid_username_facebook:
-      resp = self.app.get('/check/facebook/{}'.format(invalid_username))
-      json_resp = json.loads(resp.get_data().decode())
-      assert {
-        'possible': False,
-        'url': 'https://mbasic.facebook.com/{}'.format(invalid_username)
-      } == json_resp
-
-  def test_openhub_format_checking(self):
-    resp = self.app.get('/check/openhub/{}'.format(invalid_username))
-    json_resp = json.loads(resp.get_data().decode())
-    assert {
-      'possible': False,
-      'url': 'https://www.openhub.net/accounts/{}'.format(invalid_username)
-    } == json_resp
+  @parameterized.expand(load_test_cases('taken_username'), testcase_func_name=custom_name_func)
+  def test_taken_username(self, website, user):
+    assert_response(self.app, website, user, 200)
 
-  def test_deviantart_format_checking(self):
-    resp = self.app.get('/check/deviantart/{}'.format(invalid_username))
+  @parameterized.expand(load_test_cases('format_checking'), testcase_func_name=custom_name_func)
```

Comment:
https://travis-ci.org/manu-chroma/username-availability-checker/builds/320656667#L641

is ``re.sub`` happening in the actual data or just when creating the function name?
pls confirm this

Reply:
https://docs.python.org/3/library/re.html?highlight=re%20sub#re.sub according to python doc re.sub just "returns" not modify.